PyInstaller/hooks/hook-PySide2.QtWebEngineWidgets.py

```diff
@@ -1,51 +1,101 @@
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # Copyright (c) 2014-2019, PyInstaller Development Team.
 #
 # Distributed under the terms of the GNU General Public License with exception
 # for distributing bootloader.
 #
 # The full license is in the file COPYING.txt, distributed with this software.
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 
 import os
-from PyInstaller.utils.hooks import collect_data_files, get_qmake_path
+from PyInstaller.utils.hooks.qt import add_qt5_dependencies, \
+    pyside2_library_info
+from PyInstaller.utils.hooks import get_module_file_attribute, \
+    collect_system_data_files
+from PyInstaller.depend.bindepend import getImports
 import PyInstaller.compat as compat
 
-hiddenimports = ['PySide2.QtCore',
-                 'PySide2.QtGui',
-                 'PySide2.QtNetwork',
-                 'PySide2.QtWebChannel',
-                 'PySide2.QtWebEngineCore',
-                 ]
-
-# Find the additional files necessary for QtWebEngine.
-datas = (collect_data_files('PySide2', True, os.path.join('Qt', 'resources')) +
-         collect_data_files('PySide2', True, os.path.join('Qt', 'translations')) +
-         [x for x in collect_data_files('PySide2', False, os.path.join('Qt', 'bin'))
-          if x[0].endswith('QtWebEngineProcess.exe')])
-
-# Note that for QtWebEngineProcess to be able to find icudtl.dat the bundle_identifier
-# must be set to 'org.qt-project.Qt.QtWebEngineCore'. This can be done by passing
-# bundle_identifier='org.qt-project.Qt.QtWebEngineCore' to the BUNDLE command in
-# the .spec file. FIXME: This is not ideal and a better solution is required.
-qmake = get_qmake_path('5')
-if qmake:
-    libdir = compat.exec_command(qmake, "-query", "QT_INSTALL_LIBS").strip()
 
-    if compat.is_darwin:
-        binaries = [
-            (os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5',
-                          'Helpers', 'QtWebEngineProcess.app', 'Contents', 'MacOS', 'QtWebEngineProcess'),
-             os.path.join('QtWebEngineProcess.app', 'Contents', 'MacOS'))
-        ]
+def get_relative_path_if_possible(actual, possible_prefix):
+    actual_path = os.path.abspath(actual)
+    possible_prefix_path = os.path.abspath(possible_prefix)
```

Comment:
Is `abspath` actually useful here? It the paths are not absolute, `relpath` uses curdir as a base anyway. (And looking at the code of ´relpath` it actually calls `abspath` itself).

Reply:
Passing the paths directly into relpath now